app/harvesters/hal/hal_references_converter.py

```diff
@@ -72,7 +73,21 @@ def _abstracts(self, raw_data):
         ):
             yield Abstract(value=value, language=language)
 
-    async def _subjects(self, raw_data):
+    async def _concepts(self, raw_data):
+        fields = raw_data.get("jel_s", [])
+        # If we have jel_s fields, we use them as subjects
+        for code in fields:
+            uri = f"http://zbw.eu/beta/external_identifiers/jel#{code.split('.')[-1]}"
```

Comment:
This concatenation should be done inside the solver. Provide the JEL code as is to ConceptInformation uri (the name of the field is indeed misleading as it may contain an uri or a code) with the source indication "ConceptSources.JEL". This is enough for the ConceptFactory to choose the JelConceptSolver. JelConceptSolver will build the uri when he receives a code.